### PR TITLE
feat: Add support for IAM role and isntance profile creation

### LINKED ISSUE #192: Resource label should be specified for predefined metric type ALBRequestCountPerTarget
## Description

When you set a scaling_policy of type ALBRequestCountPerTarget you need to specify a Resource Label and this in not consider in code, so this module fail to create the resource aws_autoscaling_policy.this

- [X ] ✋ I have searched the open/closed issues and my issue is not listed.

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

- Module version 6.3.0:

- Terraform version:
Terraform v1.0.6
on linux_amd64

- Provider version(s):
provider registry.terraform.io/hashicorp/aws v4.8.0

## Reproduction Code [Required]

 no
 yes

## Expected behavior

I expect to be able to create an aws_autoscaling_policy if type ALBRequestCountPerTarget like this

```
  # Target scaling policy schedule based on Request Count per target
  scaling_policies = {
    request-count-per-target = {
      policy_type               = "TargetTrackingScaling"
      estimated_instance_warmup = 120
      target_tracking_configuration = {
        predefined_metric_specification = {
          predefined_metric_type = "ALBRequestCountPerTarget"
          resource_label         = "app/my-lb/abababababab/targetgroup/my-lb-tg/45fafafafafafa"
        }
        target_value = 1600
      }
    },
  }
```

## Actual behavior

It can't be created.

### Terminal Output Screenshot(s)
```
aws_autoscaling_policy.this["request-count-per-target"]: Creating...
╷
│ Error: Error putting scaling policy: ValidationError: Resource label should be specified for predefined metric type ALBRequestCountPerTarget
│       status code: 400, request id: 8da7c386-4988-45d3-a6cd-9974ea8c9e32
│ 
│   with aws_autoscaling_policy.this["request-count-per-target"],
│   on main.tf line 570, in resource "aws_autoscaling_policy" "this":
│  570: resource "aws_autoscaling_policy" "this" 
```

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for creating IAM role and IAM instance profile
- Fix target tracking policy which was missing the `resource_label` attribute defintion

## Motivation and Context
- IAM instance profiles are commonly associated with instances created through ASGs. This will aid in upstream efforts where this module will be used with the EKS module and the ECS module to utilize this functionality as well as other use cases
- Fixes #192 

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -180,7 +180,17 @@ module "complete" {
...
 
-  iam_instance_profile_arn = aws_iam_instance_profile.ssm.arn
+  create_iam_instance_profile = true
+  iam_role_name               = "complete-${local.name}"
+  iam_role_path               = "/ec2/"
+  iam_role_description        = "Complete IAM role example"
+  iam_role_tags = {
+    CustomIamRole = "Yes"
+  }
+  iam_role_policies = {
```
Comment (bryantbiggs) [inline examples/complete/main.tf:190]: Carrying over format from IAM so even generated IAM policies will work

File: main.tf [hunk 2]
```diff
@@ -695,6 +701,7 @@ resource "aws_autoscaling_policy" "this" {
         for_each = try([target_tracking_configuration.value.predefined_metric_specification], [])
         content {
           predefined_metric_type = predefined_metric_specification.value.predefined_metric_type
+          resource_label         = try(predefined_metric_specification.value.resource_label, null)
```
Comment (bryantbiggs) [inline main.tf:704]: This fixes #192

File: variables.tf [hunk 3]
```diff
@@ -401,12 +395,6 @@ variable "hibernation_options" {
   default     = {}
 }
 
-variable "iam_instance_profile_arn" {
```
Comment (bryantbiggs) [inline variables.tf:404]: This variable is just moved down to the IAM role/profile section

**Review by antonbabenko — APPROVED** [review]
Looks very good. Comments on specific blocks helped to understand the changes while reviewing on mobile (using github mobile app).

**Comment by antonbabenko** [comment]
This PR is included in version 6.5.0 :tada:

**Pull request merged**